subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -2832,6 +2847,132 @@ void ull_conn_default_tx_time_set(uint16_t tx_time)
 }
 #endif /* CONFIG_BT_CTLR_DATA_LENGTH */
 
+uint8_t ull_fsu_update_eff(struct ll_conn *conn)
+{
+	uint8_t fsu_changed = 0U;
+	uint8_t phy_tx;
+	uint8_t phy_rx;
+
+#if defined(CONFIG_BT_PHY_UPDATE)
+	phy_tx = conn->lll.phy_tx;
+	phy_rx = conn->lll.phy_rx;
+#else
+	phy_tx = PHY_1M;
+	phy_rx = PHY_1M;
+#endif /* CONFIG_BT_PHY_UPDATE */
+
+	fsu_changed = ull_fsu_update_eff_from_local(conn);
+
+	if (fsu_changed) {
+		/* TODO: confirm that we do not need to do something here? */
+	}
+
+	if ((conn->lll.fsu.local.spacing_type & T_IFS_CIS) == T_IFS_CIS) {
+
+		if (conn->lll.tifs_cis_us == conn->lll.fsu.eff.fsu_min) {
+			fsu_changed = 1;
+		}
+		conn->lll.tifs_cis_us = conn->lll.fsu.eff.fsu_min;
+	}
+
+	if ((conn->lll.fsu.local.spacing_type & T_IFS_ACL_CP) == T_IFS_ACL_CP) {
+		if (conn->lll.role == BT_HCI_ROLE_PERIPHERAL) {
+			if (conn->lll.fsu.local.phys & phy_tx) {
+				if (conn->lll.tifs_tx_us ==
+				    conn->lll.fsu.eff.fsu_min) {
+					fsu_changed = 1;
+				}
+				conn->lll.tifs_tx_us = conn->lll.fsu.eff.fsu_min;
+			}
+		} else {
+			if (conn->lll.fsu.local.phys & phy_rx) {
+				if (conn->lll.tifs_rx_us ==
+				    conn->lll.fsu.eff.fsu_min) {
+					fsu_changed = 1;
+				}
+				conn->lll.tifs_rx_us = conn->lll.fsu.eff.fsu_min;
+			}
+		}
+	}
+
+	if ((conn->lll.fsu.local.spacing_type & T_IFS_ACL_PC) == T_IFS_ACL_PC) {
+		if (conn->lll.role == BT_HCI_ROLE_PERIPHERAL) {
+			if (conn->lll.fsu.local.phys & phy_rx) {
+				if (conn->lll.tifs_rx_us ==
+				    conn->lll.fsu.eff.fsu_min) {
+					fsu_changed = 1;
+				}
+				conn->lll.tifs_rx_us = conn->lll.fsu.eff.fsu_min;
+			}
+		} else {
+			if (conn->lll.fsu.local.phys & phy_tx) {
+				if (conn->lll.tifs_tx_us ==
+				    conn->lll.fsu.eff.fsu_min) {
+					fsu_changed = 1;
+				}
+				conn->lll.tifs_tx_us = conn->lll.fsu.eff.fsu_min;
+			}
+		}
+	}
```

Comment:
There are a whole lot of calculations here without any documentation, explanation or references. Can you please expand on these so that it's easier for a reader to know what is going on here?